core/src/main/java/org/arquillian/smart/testing/configuration/ObjectMapper.java

```diff
@@ -0,0 +1,177 @@
+        Map<String, Object> map) {
+        method.setAccessible(true);
+        if (method.getParameterTypes().length != 1) {
+            return;
+        }
+        Class<?> parameterType = method.getParameterTypes()[0];
+        final String field = method.getName().substring(3);
+        final String property = Character.toLowerCase(field.charAt(0)) + field.substring(1);
+        Object configFileValue = map.get(property);
+
+        Optional<ConfigurationItem> foundConfigItem =
+            configItems.stream().filter(item -> property.equals(item.getParamName())).findFirst();
+
+        Object converted = null;
+        if (!foundConfigItem.isPresent()) {
+            if (!ConfigurationSection.class.isAssignableFrom(parameterType)) {
+                return;
+            } else if (configFileValue == null) {
+                converted = mapToObject((Class<ConfigurationSection>) parameterType, new HashMap<>(0));
+            } else {
+                converted =
+                    mapToObject((Class<ConfigurationSection>) parameterType, (Map<String, Object>) configFileValue);
+            }
+        } else {
+            Object mappedValue = null;
+            ConfigurationItem configItem = foundConfigItem.get();
+
+            if (configItem.getSystemProperty() != null) {
+                mappedValue = System.getProperty(configItem.getSystemProperty());
+            }
+            if (mappedValue == null) {
+                mappedValue = configFileValue;
+            }
+            if (mappedValue == null && configItem.getDefaultValue() != null) {
+                mappedValue = configItem.getDefaultValue();
+            }
+            if (mappedValue != null) {
+                converted = convert(method, mappedValue);
+            }
+        }
+
+        try {
+            if (converted != null) {
+                method.invoke(instance, converted);
+            }
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException("Failed to invoke method " + method, e);
+        }
+    }
+
+    private static Object convert(Method method, Object mappedValue) {
+        Class<?> parameterType = method.getParameterTypes()[0];
+        if (parameterType.isArray()) {
+            return handleArray(parameterType.getComponentType(), mappedValue);
+        } else if (parameterType.isEnum()) {
+            return handleEnum(method, mappedValue);
+        } else if (parameterType.isAssignableFrom(List.class)) {
+            return handleList(method, mappedValue);
+        } else if (ConfigurationSection.class.isAssignableFrom(parameterType)) {
+            if (parameterType.isAssignableFrom(mappedValue.getClass())) {
+                return mappedValue;
+            }
+            return mapToObject((Class<ConfigurationSection>) parameterType, (Map<String, Object>) mappedValue);
+        } else if (parameterType.isAssignableFrom(mappedValue.getClass())) {
+            return mappedValue;
```

Comment:
now looking at it - moving the 
```java
} else if (parameterType.isAssignableFrom(mappedValue.getClass())) {
            return mappedValue;
```
on the line 93 would eliminate the nested if on the lines 94-96

Reply:
no they both covers different scenarios. If you run `ObjectMapperTest` you will get clear idea why I did this way

Reply:
I changed it to:
```java
        if (parameterType.isArray()) {
            return handleArray(parameterType.getComponentType(), mappedValue);
        } else if (parameterType.isEnum()) {
            return handleEnum(method, mappedValue);
        } else if (parameterType.isAssignableFrom(List.class)) {
            return handleList(method, mappedValue);
        } else if (parameterType.isAssignableFrom(mappedValue.getClass())) {
            return mappedValue;
        } else if (ConfigurationSection.class.isAssignableFrom(parameterType)) {
            return mapToObject((Class<ConfigurationSection>) parameterType, (Map<String, Object>) mappedValue);
        } else {
            return convertToType(parameterType, mappedValue.toString());
        }
```
and the tests are green